File: backend/src/main/java/com/example/taskmanagement/entity/Employee.java
package com.example.taskmanagement.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

@Entity
@Table(name = "employees")
@Getter
@Setter
@ToString(exclude = {"user"})
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Employee implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @EqualsAndHashCode.Include
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "employee_id")
    private Long employeeId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @Column(name = "first_name", nullable = false, length = 50)
    private String firstName;

    @Column(name = "last_name", nullable = false, length = 50)
    private String lastName;

    @Column(name = "position", length = 50)
    private String position;

    @Column(name = "department", length = 50)
    private String department;

    @Column(name = "hired_at")
    private LocalDateTime hiredAt;

    @Column(name = "employee_group", length = 100)
    private String group;

    @Column(name = "skills", columnDefinition = "TEXT")
    private String skills;

    @PrePersist
    protected void onCreate() {
        if (hiredAt == null) {
            hiredAt = LocalDateTime.now();
        }
    }
}

File: mobile/lib/models/employee.dart
class Employee {
  final int employeeId;
  final String firstName;
  final String lastName;
  final String? position;
  final String? department;
  final String? hiredAt;

  Employee({
    required this.employeeId,
    required this.firstName,
    required this.lastName,
    this.position,
    this.department,
    this.hiredAt,
  });

  String get fullName => '$firstName $lastName'.trim();

  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
      employeeId: json['employeeId'] as int,
      firstName: json['firstName'] as String,
      lastName: json['lastName'] as String,
      position: json['position'] as String?,
      department: json['department'] as String?,
      hiredAt: json['hiredAt'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'position': position,
      'department': department,
    };
  }
}
